### PR TITLE
 implementing view::addressof (#1030) 

### PR DESCRIPTION
**Pull request by tower120**

### CODE REVIEW

File: include/range/v3/view/addressof.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                    template<class V>
+                    constexpr V *operator()(V &value) const noexcept
+                    {
+                        return std::addressof(value);
+                    }
+                    template<class V>
+                    constexpr V const *operator()(V const &value) const noexcept
+                    {
+                        return std::addressof(value);
+                    }
```
Comment (ericniebler) [inline include/range/v3/view/addressof.hpp]: This second overload is unnecessary.

File: test/view/addressof.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+using namespace ranges;
+
+void simple_test()
+{
+    std::vector<int> list = {1,2,3};
+
+    auto out = list | view::addressof;
+
+    check_equal(out, {&list[0], &list[1], &list[2]});
+}
```
Comment (ericniebler) [inline test/view/addressof.cpp]: Please also add tests for an input (not forward) range, and test also that you cannot pipe ranges with prvalue or xvalue reference types to `view::addressof`.

**Comment by ericniebler** [comment]
Looks good. One last thing: please add `#include <range/v3/view/addressof.hpp>` to `<range/v3/view.hpp>`. Then I'll merge.

**Comment by tower120** [comment]
>  One last thing: please add #include <range/v3/view/addressof.hpp> to <range/v3/view.hpp>. 

I'm sorry to ask, but - what for?
`addressof.hpp` includes `transform.hpp`. That's a ton of potentially unnecessary code for each `#include <range/v3/view.hpp>`.
Besides, `addressof.hpp` depends on `view.hpp`. This could easily cause circular dependency at some point (it does not now - due to deferred template instantiation, but such things fragile).

**Comment by CaseyCarter** [comment]
> > One last thing: please add #include <range/v3/view/addressof.hpp> to <range/v3/view.hpp>.
> 
> I'm sorry to ask, but - what for?
> `addressof.hpp` includes `transform.hpp`. That's a ton of potentially unnecessary code for each `#include <range/v3/view.hpp>`.
> Besides, `addressof.hpp` depends on `view.hpp`. This could easily cause circular dependency at some point (it does not now - due to deferred template instantiation, but such things fragile).

`<range/v3/view.hpp>` is the header that includes the definitions of all the views; `<range/v3/view/view.hpp>` is the view machinery upon which all of the view definitions (including addressof) depend.

**Comment by tower120** [comment]
@CaseyCarter Oh I see. Thanks.
@ericniebler please, update documentation as well. Online version is already differs from repository's.

**Pull request merged**